pkg/scm/git/git.go

```diff
@@ -29,6 +29,9 @@ type Git interface {
 	Checkout(repo, ref string) error
 	SubmoduleUpdate(repo string, init, recursive bool) error
 	GetInfo(string) *api.SourceInfo
+	CreateEmptyLocalGitDirectory() (string, error)
+	CreateLocalGitDirectory() (string, error)
+	CreateLocalGitDirectoryWithSubmodule() (string, error)
```

Comment:
Here are the facts as I see them:
- the 3 routines ideally ought to be grouped together given they're doing similar things
- CreateLocalGitDirectory() is called from pkg/scm/scm_test.go (different package); all the routines are called from pkg/scm/git/*_test.go.
- CreateLocalGitDirectoryWithSubmodule() calls git.ToSlashCygwin() to ensure the pathname sent to `git submodule add` is valid if running on Windows and the `git` binary is a Cygwin binary (which IMO is the likely case when running on Windows).
- git.ToSlashCygwin() is also called by non-test code.

So:
- The routines can't easily go in _test.go files because they're used across multiple packages.
- Unfortunately, putting them in pkg/test is problematic because it creates a import loop pkg/test->pkg/scm/git (pkg/scm/git already imports pkg/test).

Bearing in mind all of the above, and that these routines use the CommandRunner just like other stiGit functions use, and they are more fiddly than they look when you factor in support for testing on Windows (e.g. use of ToSlashCygwin), I opted to put them where you see them currently.  I think I preferred correctness and non-duplication of code over the cleanliness of the Git interface.